Restore archdesc/dsc check on FA gen, refs #10591

jraddaoui opened the pull request:
Removed in 0d71fb5 and causing a `java.lang.NullPointerException`
in the FOP command.

sbreker reviewed (approved):
Looks good @jraddaoui!

jraddaoui committed:
Fix FA generation, refs #10591

- Restore archdesc/dsc check, removed in 0d71fb5 and causing
  a `java.lang.NullPointerException` in the FOP command.
- Fix FA bookmarks, get head or unittitle value instead of
  aplying their templates in bookmarks as the bookmark titles
  don't accept block elements.
